lib/fog/libvirt/compute.rb

```diff
@@ -120,16 +120,16 @@ def enhance_uri(uri)
           # on macosx, chances are we are using libvirt through homebrew
           # the client will default to a socket location based on it's own location (/opt)
           # we conveniently point it to /var/run/libvirt/libvirt-sock
-          # if no socket option has been specified explicitly
+          # if no socket option has been specified explicitly and
+          # if the socket exists
 
-          if RUBY_PLATFORM =~ /darwin/
+          socketpath="/var/run/libvirt/libvirt-sock"
+          if RUBY_PLATFORM =~ /darwin/ and File.exist?("socketpath")
```

Comment:
One more and I am good, thank you:

```suggestion
          if RUBY_PLATFORM =~ /darwin/ && File.exist?("socketpath")
```